src/main/java/appeng/me/storage/ItemCellInventory.java

```diff
@@ -0,0 +1,295 @@
+			}
+		}
+
+		return false;
+	}
+
+	@Override
+	public IAEItemStack injectItems( final IAEItemStack input, final Actionable mode, final IActionSource src )
+	{
+		if( input == null )
+		{
+			return null;
+		}
+		if( input.getStackSize() == 0 )
+		{
+			return null;
+		}
+
+		if( this.cellType.isBlackListed( this.i, input ) )
+		{
+			return input;
+		}
+		// This is slightly hacky as it expects a read-only access, but fine for now.
+		// TODO: Guarantee a read-only access. E.g. provide an isEmpty() method and ensure CellInventory does not write
+		// any NBT data for empty cells instead of relying on an empty IItemContainer
+		if( ItemCellInventory.isStorageCell( input.getDefinition() ) )
+		{
+			final IMEInventory meInventory = getCell( input.getDefinition(), null );
+			if( meInventory != null && !this.isEmpty( meInventory ) )
+			{
+				return input;
+			}
+		}
+
+		final IAEItemStack l = this.getCellItems().findPrecise( input );
+		if( l != null )
+		{
+			final long remainingItemCount = this.getRemainingItemCount();
+			if( remainingItemCount < 0 )
+			{
+				return input;
+			}
+
+			if( input.getStackSize() > remainingItemCount )
+			{
+				final IAEItemStack r = input.copy();
+				r.setStackSize( r.getStackSize() - remainingItemCount );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + remainingItemCount );
+					this.updateItemCount( remainingItemCount );
+					this.saveChanges();
+				}
+				return r;
+			}
+			else
+			{
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + input.getStackSize() );
+					this.updateItemCount( input.getStackSize() );
+					this.saveChanges();
+				}
+				return null;
+			}
+		}
+
+		if( this.canHoldNewItem() ) // room for new type, and for at least one item!
+		{
+			final int remainingItemCount = (int) this.getRemainingItemCount() - this.getBytesPerType() * itemsPerByte;
+			if( remainingItemCount > 0 )
+			{
+				if( input.getStackSize() > remainingItemCount )
+				{
+					final IAEItemStack toReturn = input.copy();
+					toReturn.setStackSize( input.getStackSize() - remainingItemCount );
+					if( mode == Actionable.MODULATE )
+					{
+						final IAEItemStack toWrite = input.copy();
+						toWrite.setStackSize( remainingItemCount );
+
+						this.cellItems.add( toWrite );
+						this.updateItemCount( toWrite.getStackSize() );
+
+						this.saveChanges();
+					}
+					return toReturn;
+				}
+
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( input.getStackSize() );
+					this.cellItems.add( input );
+					this.saveChanges();
+				}
+
+				return null;
+			}
+		}
+
+		return input;
+	}
+
+	@Override
+	public IAEItemStack extractItems( final IAEItemStack request, final Actionable mode, final IActionSource src )
+	{
+		if( request == null )
+		{
+			return null;
+		}
+
+		final long size = Math.min( Integer.MAX_VALUE, request.getStackSize() );
+
+		IAEItemStack Results = null;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)